src/intrinsics/ecma262/Array.js

```diff
@@ -231,6 +234,37 @@ export default function(realm: Realm): NativeFunctionValue {
         // c. Let mapping be true.
         mapping = true;
       }
+      // If we're in pure scope and the items are completely abstract,
+      // then create an abstract temporal with an array kind
+      if (realm.isInPureScope() && items instanceof AbstractValue && items.values.isTop()) {
+        let args = [arrayFrom, items];
+        let safeToCreateTemporal = true;
+
+        // if mapfn or thisArg exist and are not statically known,
+        // we cannot guarantee that something will be not be havoced
+        // so rather, we skip this logic and continue with the default
+        // behaviur of Array.from
+        if (mapfn) {
+          if (!(mapfn instanceof ECMAScriptSourceFunctionValue || mapfn instanceof BoundFunctionValue)) {
+            safeToCreateTemporal = false;
+          }
+          args.push(mapfn);
+          if (thisArg) {
+            if (!(thisArg instanceof AbstractValue)) {
+              safeToCreateTemporal = false;
```

Comment:
Why does the thisArg matter? We know the implementation that it is going to be passed to.

Reply:
We don't support it right now, it was to be dealt with in a follow up PR.